### PR TITLE
MISUV-4289 : Remove getCurrentTaxEndYear method from IncomeSourceDetailsModel

### PR DESCRIPTION
**Pull request by AshokGitAccount**

### CODE REVIEW

File: app/audit/models/WhatYouOweResponseAuditModel.scala [hunk 1]
```diff
@@ -29,7 +29,7 @@ case class WhatYouOweResponseAuditModel(user: MtdItUser[_],
                                         whatYouOweChargesList: WhatYouOweChargesList,
                                         R7bTxmEvents: Boolean, dateService: DateServiceInterface) extends ExtendedAuditModel {
 
-  val currentTaxYear = user.incomeSources.getCurrentTaxEndYear(dateService)
+  val currentTaxYear = user.incomeSources.currentTaxYearEnd(dateService)
```
Comment (avelx) [inline app/audit/models/WhatYouOweResponseAuditModel.scala:32]: * 'currentTaxYear is redundant property which we can drop and call corresponding function from dateService where needed;

File: app/controllers/HomeController.scala [hunk 2]
```diff
@@ -140,7 +140,7 @@ class HomeController @Inject()(val homeView: views.html.Home,
       handleShowRequest(
         itvcErrorHandler = itvcErrorHandler,
         isAgent = false,
-        user.incomeSources.getCurrentTaxEndYear,
+        user.incomeSources.currentTaxYearEnd(dateService),
```
Comment (avelx) [inline app/controllers/HomeController.scala:143]: it seems that we just calling another method/property from user.incomeSources~

File: app/services/DateService.scala [hunk 3]
```diff
@@ -46,10 +46,19 @@ class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) e
       case _ => currentDate.getYear + 1
     }
   }
+
+  def getCurrentTaxEndYear: Int = {
```
Comment (avelx) [inline app/services/DateService.scala:50]: this need to be reduced to =>
getCurrentTaxYearEnd(getCurrentDate)

**Review by avelx — CHANGES_REQUESTED** [review]
minor changes requested

File: app/audit/models/WhatYouOweResponseAuditModel.scala (hunk 1 above)
Comment (AshokGitAccount) [reply app/audit/models/WhatYouOweResponseAuditModel.scala]: Done

File: app/controllers/HomeController.scala (hunk 2 above)
Comment (AshokGitAccount) [reply app/controllers/HomeController.scala]: Done

File: app/services/DateService.scala (hunk 3 above)
Comment (AshokGitAccount) [reply app/services/DateService.scala]: Removed getCurrentTaxEndYear as we already had a method returning the same thing

File: app/models/incomeSourceDetails/IncomeSourceDetailsResponse.scala [hunk 4]
```diff
@@ -42,22 +42,18 @@ case class IncomeSourceDetailsModel(mtdbsa: String,
...
   val hasPropertyIncome: Boolean = property.nonEmpty
   val hasBusinessIncome: Boolean = businesses.nonEmpty
 
-  def getCurrentTaxEndYear(implicit dateService: DateServiceInterface) : Int = {
-    val currentDate = dateService.getCurrentDate
-    if (currentDate.isBefore(LocalDate.of(currentDate.getYear, 4, 6))) currentDate.getYear
-    else currentDate.getYear + 1
-  }
-
+  val currentTaxYearEnd: DateServiceInterface => Int =
```
[the remark this answers is not in the document]
Comment (AshokGitAccount) [reply app/models/incomeSourceDetails/IncomeSourceDetailsResponse.scala]: done

File: app/services/DateService.scala [hunk 5]
```diff
@@ -46,10 +46,12 @@ class DateService @Inject()(implicit val frontendAppConfig: FrontendAppConfig) e
...
     }
   }
+
 }
 
 trait DateServiceInterface {
   def getCurrentDate: LocalDate
 
   def getCurrentTaxYearEnd(currentDate: LocalDate): Int
+
```
Comment (avelx) [inline app/services/DateService.scala:56]: it is actually fine to have extra method you created originally, idea was just to redesign things a bit

**Review by avelx — APPROVED** [review]
looks good

**Pull request merged**